tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFile.java

```diff
@@ -488,6 +494,36 @@ public String getBaileyValue(String xpath, Output<String> pathWhereFound, Output
         return result;
     }
 
+    /**
+     * Similar to {@link #getBaileyValue(String, Output, Output)} except returns a map, as constructed
+     * values can have multiple contributory paths.
+     * @return map from xpath to locale of one or more contributing paths in the case of glossonymns or other
+     * constructed values.
+     */
+    public List<LocaleInheritanceInfo> getPathsWhereFound(String xpath) {
+        List<LocaleInheritanceInfo> list = new LinkedList<>();
+        // first, call getBaileyValue
+        Status status = new Status();
+        final String locale1 = getSourceLocaleIdExtended(xpath, status, true, list);
+        final String path1 = status.pathWhereFound;
```

Comment:
The comment says "first, call getBaileyValue" -- and yet, getBaileyValue isn't called here at least not directly

The relationship between getSourceLocaleIdExtended and getBaileyValue is complicated

Note that CLDRFile.getBaileyValue calls getSourceLocaleID

So I'm in doubt what the PathInfo class/tool is really showing, or should be showing -- my impression is that CLDRFile.getBaileyValue is the single most important function for us to be able to explain

Note that XMLSource.ResolvingSource.getBaileyValue is very distinct from CLDRFile.getBaileyValue and the latter is generally the final decider of inheritance at least as far as Survey Tool is concerned...

Reply:
it's a leftover. I actually didn't want to call getBaileyValue here.

Reply:
This sounds like you're wanting a _Bailey_ explainer. The ticket was originally setup to attempt to explain where an actual value's inheritance came from.

Reply:
anyway, i think a Bailey explainer could probably work similarly

Reply:
yes, I think we need a Bailey explainer, above all

could it be, that an inheritance explainer and a bailey explainer are exactly the same thing if the value is inherited

and if the value isn't inherited, there's nothing to explain?

Reply:
let's discuss tomorrow… in any event, right now if you run this where there's an exact value it will tell you there's a value… if it's inherited, it will explain the inheritance.

Reply:
that has been my (mis-?)understanding...

now we need a bailey explainer explainer, and a bailey explainer explainer explainer, ad infinitum

Reply:
seriously, though ... even when a value is not inherited, we still need to know the bailey value, and explain the bailey value

Reply:
So look at this comment => https://github.com/unicode-org/cldr/pull/2860#issuecomment-1512108214 Is that what you'd expect from this tool or something different? (Assuming it explains _how_ the inheritance got that way, which it doesn't right now)

Reply:
Can we start with an inheritance explainer and add Bailey in the next round?

Reply:
They are _very_ closely related. The bailey value is precisely what would be inherited if a value didn't exist.
The only difference would be that inheritance stops when the first hard value is found.